### PR TITLE
Update instructions and add a license

### PR DESCRIPTION
**Pull request by therealkujo**
## tl;dr
I went through this bootcamp recently and noticed some areas where the instructions may not be as clear or where we are using outdated actions modules so I am updating the bootcamp to reflect the latest changes.

This repo is also lacking a license so I am adding the MIT license.

I ran these changes by @s-samadi and she recommended I open a PR.

### README

In the README, the instructions were to use the template to create a repo in the `ghas-bootcamp` repo or to just create the repo and push the contents of this repo to it. Since I did not have access to this repo, I made a personal repo but in doing so you cannot fully complete the labs because things like the secret scanning dashboard is not available for public repos. I added a clarification in there that if we are creating the repo somewhere outside the org `ghas-bootcamp` they should do it in an org with GHAS enabled to be able to fully leverage this lab.

### Lab 2 - Secret Scanning Configuration

In this lab, we are asked to create a `secret_scanning.yml` to ignore some file paths so that we can experiment with pushing secrets in those files to see what the behaviour is like. While this is poorly documented, I had a discussion with the secret scanning team and confirmed with them that if a user creates a `secret_scanning.yml` and then pushes a file that is listed in `paths-ignore`, the secret scanning will still scan the file. 

This is actually expected behaviour because there exists a scenario that a developer can add new paths to be ignored and then leak tokens in those files to the repo without the repo admins agreeing and approving the changes to the ignore list first. The `secret_scanning.yml` must be merged to the default branch first before a developer can upload tokens into those files that are in the `secret_scanning.yml` to have them ignored. This applies to existing tokens already found and for push protection.

### Lab 3 - Code Scanning Configuration

We are asking users to add the `setup-java` actions step and in our example we tell them to use `actions/setup-java@v1` but if a user is adding this in the UI using the help prompts, the latest version for `actions/setup-java` is actually `3.4.1` or `v3`.  Support for Java 15 is dropped and so I am upping the version to 16. I did verify that the codebase will compile and that code scanning will complete with using Java 16.

The other addition is the `distribution` field which asks us to specify what kind of jdk we want to use. For simplicity, I chose `microsoft` and verified that the build will run to completion using these configurations. For more information, you can take a look at [actions/setup-java](https://github.com/actions/setup-java) repo.  

### Codeql CLI

I updated the documentation to get users to leverage the GitHub CLI to install and manage the CodeQL CLI. The previous documentation had users downloading releases and having to play with their PATH variable which for some people they may be uncomfortable with and it also introduces issues when you start downloading multiple versions of the cli and bundle and need to manage it. Having a non-standard installation method isn't a best practice. The GitHub CLI will install CodeQL into a more standard installation spot and managing the bundles will not require any PATH manipulations. It also helps to simplify the code scanning part as we no longer need to reference the full path for a query suite but we can now use relative paths.

### CODE REVIEW

**Comment by therealkujo** [comment]
Tagging @JenniferKerns for review
